spec.bs

```diff
@@ -1222,6 +1222,64 @@ Each {{FencedFrameConfig}} has:
   1. Initialize |value|'s [=fencedframeconfig/contentHeight=] to |serialized|.\[[ContentHeight]].
 </div>
 
+To help with ease of adoption,
+[until 2026](https://github.com/WICG/turtledove/issues/286#issuecomment-1682842636) we will support
+the API {{Window/navigator}}.{{Navigator/deprecatedReplaceInURN()}}, which allows you to substitute
+macros into the [=fenced frame config/mapped url=] corresponding to a given [=urn uuid=] or
+{{FencedFrameConfig}}.
+
+<pre class=idl>
+typedef (USVString or FencedFrameConfig) UrnOrConfig;
+
+partial interface Navigator {
+  undefined deprecatedReplaceInURN(
+    UrnOrConfig urnOrConfig, record&lt;USVString, USVString&gt; replacements);
+};
+</pre>
+
+<div algorithm>
+  To <dfn export for=fencedframeutil>substitute macros</dfn> with an [=ordered map=] with
+  [=string=] [=map/keys=] and [=string=] [=map/values=] |macros| into a [=string=] |string|, run
+  these steps:
+
+  1. <span class=XXX>TODO:</span> [Spec this](https://github.com/WICG/fenced-frame/issues/116).
+     Substitute the keys from |macros| with the corresponding values into |string|, and return the
+     new string. There should be no recursive substitution.
+</div>
+
+<div algorithm>
+  The <dfn method for=Navigator>deprecatedReplaceInURN(|urnOrConfig|, |replacements|)</dfn>
+  method steps are:
+
+  1. Let |urn| be null.
+
+  1. If |urnOrConfig| is a {{USVString}}, set |urn| to |urnOrConfig|.
+
+  1. Otherwise, set |urn| to |urnOrConfig|'s [=fencedframeconfig/urn=].
+
+  1. Let |mapping| be [=this=]'s [=relevant global object=]'s [=associated Document=]'s
+     [=node navigable=]'s [=navigable/traversable navigable=]'s
+     [=traversable navigable/fenced frame config mapping=].
+
+  1. Let |config| be the result of [=fenced frame config mapping/finding a config=] in |mapping|
```

Comment:
It seems wrong to do this synchronously, and seems to not match what the implementation does, since the implementation does this "in parallel" and returns a promise.

Reply:
Yeah, I should make this a promise...

Reply:
Turned this into a promise, modeling after some parts of the Protected Audience spec. Lmk if it I didn't do it right.